ckanext/embeddings/actions.py

```diff
@@ -12,18 +12,17 @@ def package_similar_show(context, data_dict):
     except ValueError:
         raise toolkit.ValidationError(f"Wrong value for limit paramater: {limit}")
 
+    field_name = toolkit.config.get("ckanext.embeddings.solr_vector_field_name", "vector")
+
     try:
-        dataset_dict = toolkit.get_action("package_show")(
-            {"ignore_auth": True}, {"id": dataset_id}
-        )
-    except toolkit.ObjectNotFound:
+        vectors = toolkit.get_action("package_search")(
+            {"ignore_auth": True}, {"fq": f"(id:{dataset_id} OR name:{dataset_id})", 'fl':f"{field_name},id"}
```

Comment:
Nice one! I always forget that we allow custom `fl` values this makes sense

Reply:
One thing I'm thinking about here is that pulling the vector along with the package would give us the ability to calculate the similarity in python, and then potentially use it for thresholding or reranking, or merging with a keyword search.  On the other hand, my explorations generally show that there's a set of hits for "we have no idea what this search means" and the similarities are quite good, so it's not necessarily going to be easy to understand the similarity scores.